### DISCUSSION TITLE
rmsd in viamd?

### DISCUSSION THREAD

**Discussion by Gorynya** [scanberg/viamd#64] [Q&A]
Hello everyone! Can someone give an example of how to use the rmsd function in a viamd program? I can't figure out how to specify the reference MD trajectory frame or the reference structure, for example? I liked the program as a whole very much, the idea is wonderful. But there is an acute shortage of available functions, or may be I just could not find them, so far I mainly use VMD with the ability to write scripts there to analyze MD trajectories. Thank you in advance for your answers!

**Comment by mathieulinares** [comment]
Thanks for your message. 
The rmsd is calculated using the first frame of the trajectory as a reference structure.
Here is some example:
r = rmsd(all); # will calculate the rmsd of your whole system.
r1 = rmsd(resname("MOL")[1]); will calculate the rmsd of the first residue named MOL
r2 = rmsd(all) in resname("SOL"); will calculate the rmsd for all your water molecules.

Regarding functions available. There are not yet ALL listed in the language page of the wiki. I will try to update this page ASAP.
You can find examples of the functions available in the tutorials and in the youtube videos. 

Do not hesitate to ask what type of functions you are looking for as a reply of this message.

**Reply by Gorynya** [reply]
Thanks a lot for the reply! I'm going to try it for my trajectories now. About adding functionality. Well, in fact, if we start from the VMD functions, then all the commands of the measure group would be very desirable. Personally, I really need the commands "measure contacts", "measure rmsd", "measure rmsf".
https://www.ks.uiuc.edu/Research/vmd/current/ug/node138.html
At the same time, it is important to be able to choose the reference sets of atoms yourself, whether it is a specific frame of the trajectory or a completely different structure. In addition, I would like to be able to align the trajectory frames according to the selected set of atoms. Perhaps it would not be superfluous to have an automatic procedure that determines 3 the most stable atoms and aligns the trajectory along them. So far, as far as I understand, there is an automatic alignment to the center of mass, which is not very convenient, it seems to me. Thank you very much and good luck in program development!
P.S.
Perhaps it is already possible to align frames on certain atoms or atom sets? And besides, I'm a little confused, but how do I save the structure (atomic coordinates) from a certain trajectory frame?
P.P.S.
One more, may be silly question.
About Distribution window, what are the data for the Y-axis? For the X-axis it is the rmsd value in angstroms, right? Thank you!
![test](https://github.com/scanberg/viamd/assets/160626099/9e96ddd0-5c4e-433f-91f8-55ff5ba59300)
